remotes/docker/resolver.go

```diff
@@ -65,6 +66,9 @@ type ResolverOptions struct {
 	// is interpretted as a long lived token.
 	Credentials func(string) (string, string, error)
 
+	// Host is the hostname of the registry.
+	Host string
```

Comment:
Can we provide this as a function? A resolver is intended to handle multiple namespaces/hosts

We could have the `docker.io` handling as a default function here

Reply:
Will do.

Reply:
How about this?